### PR TITLE
Fix multiple issues with FinalizationRegistry content

### LINKED ISSUE #13695: Issue with "FinalizationRegistry.prototype.register()": possible values for `heldValue` aren't mentioned
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry/register

#### What information was incorrect, unhelpful, or incomplete?

I thought I can only have a string as `heldValue` but apparently it can be any object. I'd appreciate some emphasis on that as `The value` is a bit ambiguous.

#### Specific section or headline?

> heldValue
>  The value to pass to the finalizer for this object. This cannot be the target object.

#### What did you expect to see?

> heldValue
>  The value to pass to the finalizer for this object. This cannot be the target object but any other object (including functions) or primitive value.

#### Did you test this? If so, how?
Using the tests at https://www.javascripture.com/FinalizationRegistry, changing the `42` to `new WeakRef({})` still printed

```
Allocating a lot of objects to try to force garbage collection
cleanup: [object WeakRef]
```
but not sure if it also allows functions, I made that up.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/finalizationregistry/register`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry/register
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/finalizationregistry/register/index.md
* Last commit: https://github.com/mdn/content/commit/cf607d68522cd35ee7670782d3ee3a361eaef2e4
* Document last modified: 2021-07-20T18:47:57.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

Fix #13695

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md [hunk 1]
```diff
@@ -23,12 +23,11 @@ unregister(unregisterToken);
...
 
 - `unregisterToken`
-  - : The token used with the {{jsxref("FinalizationRegistry.prototype.register",
-        "register")}} method when registering the target object.
+  - : The token used with the {{jsxref("FinalizationRegistry.prototype.register", "register")}} method when registering the target object. Multiple cells registered with the same `unregisterToken` will be unregistered together.
 
 ### Return value
 
-`undefined`.
+`true` if at least one cell was unregistered; `false` if no cell was unregistered.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md:30]:
```suggestion
A boolean value that is `true` if at least one cell was unregistered, `false` if no cell was unregistered.
```

Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md:30]: In addition to the boolean value, I think there are cases where `undefined` can be returned, am I wrong?

File: files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md [hunk 2]
```diff
@@ -23,12 +23,11 @@ unregister(unregisterToken);
...
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md:31]: I think we have cases where the `{{jsxref("TypeError")}}` exception is raised. We likely want an `### Exceptions` section.

File: files/en-us/web/javascript/reference/global_objects/finalizationregistry/register/index.md [hunk 3]
```diff
@@ -27,8 +27,7 @@ register(target, heldValue, unregisterToken);
 - `target`
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/finalizationregistry/register/index.md:27]: I think we have cases where the `{{jsxref("TypeError")}}` exception is raised. We likely want an `### Exceptions` section.

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Thanks a lot.

I think there are extra things that should be updated. What do you think?

File: files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md:30]: [Can't find it in spec](https://tc39.es/ecma262/#sec-finalization-registry.prototype.unregister). It either returns a boolean, or throws a `TypeError`.

File: files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md (hunk 2 above)
Comment (Josh-Cena) [reply files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md:31]: Makes sense—when I work on these docs there are a lot of sections missing mentions of exceptions. Is there a convention for how this should be structured? Or just an h3 section.

Comment (teoli2003) [reply files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md:35]: Yes, we should add them as we do for API, with a definition list in that section: https://developer.mozilla.org/en-US/docs/MDN/Structures/Page_types/API_method_subpage_template#exceptions

Something like: https://developer.mozilla.org/en-US/docs/Web/API/IDBCursor/continue#exceptions

And yes, there are a lot of exceptions not listed (both under JS and API), I would like to make a full check once, but I don't have the time. Getting the exception list, with causes, complete under JavaScript/ and API/ is hard, though as they are buried inside the prose, or the algorithm, of the specifications.

File: files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md [hunk 4]
```diff
@@ -27,7 +27,11 @@ unregister(unregisterToken);
 
 ### Return value
 
-`true` if at least one cell was unregistered; `false` if no cell was unregistered.
+A boolean value that is `true` if at least one cell was unregistered and `false` if no cell was unregistered.
+
+### Exceptions
+
+A `TypeError` is thrown if `unregisterToken` is not an object.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/finalizationregistry/unregister/index.md:34]:
```suggestion
- {{jsxref("TypeError")}}
  - : Thrown when `unregisterToken` is not an object.
```

**Review by teoli2003 — APPROVED** [review]
Thank you very much. I've listed the three conditions for an exception as a list (we usually do this).

Let's merge this. If we discover more problems we always can fix them in follow-ups!

**Pull request merged**